packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -59,10 +59,15 @@ function ToggleGroupControl(
 		baseId,
 		state: value,
 	} );
+	const previousValue = usePrevious( value );
 
 	// Propagate radio.state change
 	useUpdateEffect( () => {
-		onChange( radio.state );
+		// Avoid calling onChange if radio state changed
+		// from incoming value.
+		if ( previousValue !== radio.state ) {
```

Comment:
It feels like this should be part of useUpdateEffect instead?

Reply:
Why? It seems that `useUpdateEffect` just skips first render, but here we can pass an incoming value later on..

Reply:
I guess I was confused by the "useUpdateEffect" name, for me it was about syncing and calling an onChange value from an internal state but I guess it's not :P

Reply:
I was confused at first as well by the name 😄